static/js/util/util.js

```diff
@@ -168,30 +169,58 @@ export function makeCourseProgressDisplay(course, isFirst, isLast) {
  * @param {Object} profile The user profile
  * @returns {Object} Validation errors or an empty object if no errors
  */
-export function validateProfile(profile) {
+export function validateProfile(profile, requiredFields, messages) {
+  let clone = _.cloneDeep(profile);
+  let fields = Object.entries(clone).filter(pair => _.isObject(pair[1]));
+  let nullObj = (obj) => {
+    for (let key in obj) { obj[key] = null };
+  }
+  fields.map( (pair) => {
+    for (let obj of pair[1]) {
+      nullObj(obj);
+    }
+  });
   let errors = {};
+  fields.map( pair => errors[pair[0]] = pair[1]);
 
-  let required = {
-    'first_name': "Given name",
-    'last_name': "Family name",
-    'preferred_name': "Preferred name",
-    'gender': "Gender",
-    'preferred_language': "Preferred language",
-    'city': "City",
-    'country': "Country",
-    'birth_city': 'City',
-    'birth_country': "Country",
-    'date_of_birth': "Date of birth"
-  };
+  for (let keySet of requiredFields) {
+    if (!dig(profile, keySet)) {
```

Comment:
lodash includes `_.get`, which I'm using instead now. It's a little bit better behaved.